src/main/java/ecdar/backend/BackendDriver.java

```diff
@@ -92,42 +91,42 @@ private BackendConnection getBackendConnection(BackendInstance backend) throws B
     }
 
     /**
-     * Attempts to start a new connection to the specified backend. On success, the backend is added to the associated
+     * Attempts to start a new connection to the specified engine. On success, the engine is added to the associated
      * queue, otherwise, nothing happens.
      *
-     * @param backend the target backend for the connection
+     * @param engine the target engine for the connection
      */
-    private void tryStartNewBackendConnection(BackendInstance backend) {
+    private void tryStartNewEngineConnection(Engine engine) {
         Process p = null;
-        String hostAddress = (backend.isLocal() ? "127.0.0.1" : backend.getBackendLocation());
+        String hostAddress = (engine.isLocal() ? "127.0.0.1" : engine.getEngineLocation());
         long portNumber = 0;
 
-        if (backend.isLocal()) {
+        if (engine.isLocal()) {
             try {
-                portNumber = SocketUtils.findAvailableTcpPort(backend.getPortStart(), backend.getPortEnd());
+                portNumber = SocketUtils.findAvailableTcpPort(engine.getPortStart(), engine.getPortEnd());
             } catch (IllegalStateException e) {
                 // No port was available in range, we assume that connections are running on all ports
```

Comment:
This case is used to account for no connections being available and no more free ports. This means that we will just have to wait until a connection becomes available, which IMO is not something that the user should be notified of. My reasoning is mainly that the toast would be shown every time a query is requested to be executed but no connections are currently available and that this would just confuse the user, who might think something is wrong.

Maybe I should extend/rewrite the comment to explain this? I.e. 
`// No port was available in the specified port range, do not start a new connection but wait for one to become available instead`

Reply:
I think that the doc string explains this behavior, but I think that the method could be refactored to increase readability. I will do this in a separate PR

Reply:
> This case is used to account for no connections being available and no more free ports. This means that we will just have to wait until a connection becomes available, which IMO is not something that the user should be notified of. My reasoning is mainly that the toast would be shown every time a query is requested to be executed but no connections are currently available and that this would just confuse the user, who might think something is wrong.

You have a good point. I think we should let it be as is.